src/main/webapp/app/shared/layouts/navbar/navbar.component.ts

```diff
@@ -64,12 +73,31 @@ export class NavbarComponent implements OnInit, OnDestroy {
             .getAuthenticationState()
             .pipe(tap((user: User) => (this.currAccount = user)))
             .subscribe();
+
+        this.routerEventSubscription = this.router.events.subscribe((event) => {
+            if (event instanceof NavigationEnd) {
+                this.exam = undefined;
+                const examRoute = new RegExp('exams/[0-9]');
+                if (examRoute.test(event.url) && !event.url.includes('management')) {
```

Comment:
I am not an Angular Export but this whole part seems extremely brittle. Anybody got an idea how to do this whole part in a more robust way than regexing and splitting?

Reply:
usually you can access the parameters by using 
`this.route.params.subscribe((params) => {
            courseId = params['courseId']);
            examId = params['examId']);
            });`
but that didn´t work for me, this way both IDs remained undefined, no matter what..
possibly due to being inside the router event subscription, idk..

Reply:
![](https://miro.medium.com/max/2000/1*3VIOIjGN7BvgCe-KvxEuvA.png)
Following this Article [How To Get Route Path Parameters In Non-Routed Angular Components 🦀](https://medium.com/@tomastrajan/how-to-get-route-path-parameters-in-non-routed-angular-components-32fc90d9cb52). this should be possible in a slightly nicer way.

Reply:
this works, thanks 😁